src/main/java/org/tinyradius/util/RadiusClient.java

```diff
@@ -333,6 +333,8 @@ public RadiusPacket communicate(RadiusPacket request, int port) throws IOExcepti
 				// TODO increase Acct-Delay-Time by getSocketTimeout()/1000
 				// this changes the packet authenticator and requires packetOut to be
 				// calculated again (call makeDatagramPacket)
+			} finally {
+				socket.close();
```

Comment:
You may want to take into account the retry logic above.

Reply:
OK，I will fix it and pull request again.